utils/yoke/src/zerovec_impls.rs

```diff
@@ -0,0 +1,89 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::*;
+use std::mem;
+use zerovec::map::ZeroMapKV;
+use zerovec::ule::*;
+use zerovec::{VarZeroVec, ZeroMap, ZeroVec};
+
+// This impl is similar to the impl on Cow and is safe for the same reasons
+unsafe impl<'a, T: 'static + AsULE + ?Sized> Yokeable<'a> for ZeroVec<'static, T> {
+    type Output = ZeroVec<'a, T>;
+    fn transform(&'a self) -> &'a ZeroVec<'a, T> {
+        self
+    }
+
+    unsafe fn make(from: ZeroVec<'a, T>) -> Self {
+        debug_assert!(mem::size_of::<ZeroVec<'a, T>>() == mem::size_of::<Self>());
+        let ret = mem::transmute_copy(&from);
+        mem::forget(from);
+        ret
+    }
+
+    fn with_mut<F>(&'a mut self, f: F)
+    where
+        F: 'static + for<'b> FnOnce(&'b mut Self::Output),
+    {
+        unsafe { f(mem::transmute::<&mut Self, &mut Self::Output>(self)) }
+    }
+}
+
+// This impl is similar to the impl on Cow and is safe for the same reasons
+unsafe impl<'a, T: 'static + AsVarULE> Yokeable<'a> for VarZeroVec<'static, T> {
+    type Output = VarZeroVec<'a, T>;
+    fn transform(&'a self) -> &'a VarZeroVec<'a, T> {
+        self
+    }
+
+    unsafe fn make(from: VarZeroVec<'a, T>) -> Self {
+        debug_assert!(mem::size_of::<VarZeroVec<'a, T>>() == mem::size_of::<Self>());
+        let ret = mem::transmute_copy(&from);
+        mem::forget(from);
+        ret
+    }
+
+    fn with_mut<F>(&'a mut self, f: F)
+    where
+        F: 'static + for<'b> FnOnce(&'b mut Self::Output),
+    {
+        unsafe { f(mem::transmute::<&mut Self, &mut Self::Output>(self)) }
+    }
+}
+
+unsafe impl<
+        'a,
+        K: 'static + ZeroMapKV<'static> + Yokeable<'a>,
```

Comment:
What is the purpose of this `'static`?

Reply:
All Yokeables are `'static`, I'm duplicating the bound so that it's clear in this context that it's `'static` without having to remember "all yokeables are static"

Reply:
+1